### PR TITLE
NEW : Nature of product is now a dictionay

### PR DESCRIPTION
**Pull request by FHenry**
NEW : Nature of product is now a dictionay

### CODE REVIEW

File: htdocs/product/class/product.class.php [hunk 1]
```diff
@@ -4562,10 +4562,21 @@ public function getLibFinished()
 		global $langs;
 		$langs->load('products');
 
-		if ($this->finished == '0') { return $langs->trans("RowMaterial");
-		}
-		if ($this->finished == '1') { return $langs->trans("Finished");
+		if (isset($this->finished) && $this->finished>=0) {
+			$sql = 'SELECT label, code FROM '.MAIN_DB_PREFIX.'c_product_nature where code='.$this->finished.' AND active=1';
```
Comment (eldy) [inline htdocs/product/class/product.class.php:4566]: this->finished is a string or an int ?

File: htdocs/install/mysql/data/llx_c_product_nature.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+--
+-- Do not place a comment at the end of the line, this file is parsed when
+-- from the install and all '--' are removed.
+--
+-- Ne pas placer de commentaire en fin de ligne, ce fichier est parsé lors
+-- de l'install et tous les sigles '--' sont supprimés.
+--
+
+INSERT INTO llx_c_product_nature (code, label, active) VALUES (-1, '', 1);
```
Comment (eldy) [inline htdocs/install/mysql/data/llx_c_product_nature.sql:26]: Some old dictionnary has this but it is a bad practive to add a value empty into list of possible value.
This must instead addressed when we suggest the combo list by adding a paramater "$showempty".

File: htdocs/product/class/product.class.php (hunk 1 above)
Comment (FHenry) [reply htdocs/product/class/product.class.php]: @eldy Done

**Comment by FHenry** [comment]
@eldy don't understand why travis unit test DateLibTest is wrong.
I look into git history of impacted method, didn't find any clue about this, but for sure ot's not related with this PR.

**Pull request merged**